src/python/zquantum/core/circuit/conversions/symbolic_expressions_test.py

```diff
@@ -0,0 +1,246 @@
+        self, sympy_addition
+    ):
+        assert not is_addition_of_negation(sympy_addition)
+
+    @pytest.mark.parametrize(
+        "sympy_addition, expected_args",
+        [
+            (sympy.Symbol("x") - sympy.Symbol("y"), (Symbol("x"), Symbol("y"))),
+            (1 - sympy.Symbol("x"), (1, Symbol("x"))),
+        ],
+    )
+    def test_add_resulting_from_subtraction_is_converted_to_sub_function_call(
+        self, sympy_addition, expected_args
+    ):
+        assert expression_from_sympy(sympy_addition) == FunctionCall(
+            "sub", expected_args
+        )
+
+    @pytest.mark.parametrize(
+        "sympy_power, expected_args",
+        [
+            (sympy.Pow(sympy.Symbol("x"), 2), (Symbol("x"), 2)),
+            (sympy.Pow(2, sympy.Symbol("x")), (2, Symbol("x"))),
+            (
+                sympy.Pow(sympy.Symbol("x"), sympy.Symbol("y")),
+                (Symbol("x"), Symbol("y")),
+            ),
+        ],
+    )
+    def test_sympy_pow_is_converted_to_pow_function_call(
+        self, sympy_power, expected_args
+    ):
+        assert expression_from_sympy(sympy_power) == FunctionCall("pow", expected_args)
+
+    @pytest.mark.parametrize(
+        "sympy_power, expected_denominator",
+        [
+            (sympy.Pow(sympy.Symbol("x"), -1), Symbol("x")),
+            (
+                sympy.Pow(
+                    sympy.Add(sympy.Symbol("x"), sympy.Symbol("y"), evaluate=False), -1
+                ),
+                FunctionCall("add", (Symbol("x"), Symbol("y"))),
+            ),
+        ],
+    )
+    def test_sympy_power_with_negative_one_exponent_gets_converted_to_division(
+        self, sympy_power, expected_denominator
+    ):
+        assert expression_from_sympy(sympy_power) == FunctionCall(
+            "div", (1, expected_denominator)
+        )
+
+    @pytest.mark.parametrize(
+        "sympy_function_call, expected_function_call",
+        [
+            (sympy.cos(2), FunctionCall("cos", (2,))),
+            (sympy.sin(sympy.Symbol("theta")), FunctionCall("sin", (Symbol("theta"),))),
+            (sympy.exp(sympy.Symbol("x")), FunctionCall("exp", (Symbol("x"),))),
+        ],
+    )
+    def test_sympy_function_calls_are_converted_to_function_call_object_with_appropriate_function_name(
+        self, sympy_function_call, expected_function_call
+    ):
+        assert expression_from_sympy(sympy_function_call) == expected_function_call
+
+
+@pytest.mark.parametrize(
+    "sympy_expression, quil_expression",
+    [
+        (sympy.Symbol("theta"), quil.Parameter("theta")),
+        (
+            sympy.Mul(sympy.Symbol("theta"), sympy.Symbol("gamma"), evaluate=False),
+            quil.Parameter("theta") * quil.Parameter("gamma"),
+        ),
+        (sympy.cos(sympy.Symbol("theta")), quilatom.quil_cos(quil.Parameter("theta"))),
+        (sympy.cos(2 * sympy.Symbol("theta")), quilatom.quil_cos(2 * quil.Parameter("theta")))
+    ]
```

Comment:
Great idea. I added some tests in 0157e49
I had to add some additional detection of sqrt because SymPy converts it to pow(..., 0.5), but it should work no and the tests were added here: f5de537
There is no `quil_tan` function but I defined it in `QUIL_DIALECT` using sine and cosine. Implementation and tests were added here: c71e207

Note that some new tests cases once again use e.g. `sympy.Add` explicitly, that's because it's quite possible that otherwise SymPy would massacre the order of arguments (resulting in equivalent, but not an equal expression).